src/Poll.java

```diff
@@ -0,0 +1,112 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import org.pircbotx.Channel;
+import org.pircbotx.hooks.ListenerAdapter;
+import org.pircbotx.hooks.events.MessageEvent;
+
+public class Poll extends ListenerAdapter {
+
+	// Command that starts poll
+	public static final String commandKey = "!poll";
+	boolean pollEnabled = false;
+
+	// Store options
+	Map<Integer, String> pollOptions = new HashMap<>();
+	int[] votes;
+
+	// Helper to send messages to the server
+	public void sendMessage(MessageEvent event, String message) {
+		event.getChannel().send().message(message);
+	}
+
+	public boolean shouldStartPoll(MessageEvent event) {
+		if (event.getUser().getNick().equals(Constants.BROADCASTER)
+				&& event.getMessage().toLowerCase().startsWith(commandKey) && !pollEnabled) {
+			pollEnabled = true;
+			return true;
+		}
+		return false;
+	}
+
+	public boolean shouldEndPoll(MessageEvent event) {
+		if (event.getUser().getNick().equals(Constants.BROADCASTER)
+				&& event.getMessage().toLowerCase().startsWith(commandKey) && pollEnabled) {
+			pollEnabled = false;
+			return true;
+		}
+		return false;
+	}
+
+	public void displayResults(Channel channel) {
+		channel.send().message("Results:");
+
+		for(int i = 0; i < votes.length; i++) {
+			channel.send().message(pollOptions.get(i) + ": " + votes[i]);
+		}
+
+		channel.send().message("Poll has ended!");
+		// Reset poll options
+		pollOptions.clear();
+	}
+
+	public void onMessage(MessageEvent event) {
+		String message = event.getMessage();
+
+		if (shouldStartPoll(event)) {
+			String[] params = message.split(" ");
+
+			if (params.length > 1) {
+				votes = new int[params.length - 1];
+
+				for (int i = 1; i < params.length; i++) {
+					pollOptions.put(i-1, params[i]);
+				}
+
+				sendMessage(event, getPollOptions(params));
+			} else {
+				pollEnabled = false;
+				return;
+			}
+		} else if (pollEnabled) {
+			if (shouldEndPoll(event)) {
+				sendMessage(event, getPollResults());
+			}
+
+			int voteKey = 0;
+			try {
+			  voteKey = Integer.parseInt(event.getMessage());
+			} catch (NumberFormatException e) {
+			  // not a valid vote
+			}
+
+			if (voteKey != 0) {
+				votes[voteKey-1] = votes[voteKey-1] + 1;
```

Comment:
This throws an ArrayIndexOutOfBoundsException for negative values or values greater than the number of options (which happens quite a lot in twitch chat). The exception is caught and not logged or anything but you should still probably do bounds checking to handle that a little more gracefully.